magicctapipe/scripts/lst1_magic/lst1_magic_event_coincidence.py

```diff
@@ -182,6 +183,121 @@ def event_coincidence(input_file_lst, input_dir_magic, output_dir, config):
         df_magic["timestamp"] = timestamps_magic.to_value("s")
         df_magic.drop(["time_sec", "time_nanosec"], axis=1, inplace=True)
 
+        # Pre offset search is performed to define the offset scan region.
+        # First, N events are extracted from largerst intensity events for LST and
+        # MAGIC. Then, it counts a number of coincident events within a defined
+        # window after shifting all possible combination (N x N) of time offsets.
+        if pre_offset_search:
+            logger.info(
+                "\nPre offset search using large-intensity shower events is ongoing..."
+            )
+
+            logger.info(
+                f"\nExtracting the {tel_name} events taken when LST-1 observed for pre offset search..."
+            )
+
+            time_lolim = timestamps_lst[0] - window_half_width
+            time_uplim = timestamps_lst[-1] + window_half_width
+
+            cond_lolim = timestamps_magic >= time_lolim
+            cond_uplim = timestamps_magic <= time_uplim
+
+            mask_lst_obs_window = np.logical_and(cond_lolim, cond_uplim)
+            n_events_magic = np.count_nonzero(mask_lst_obs_window)
+
+            if n_events_magic == 0:
+                logger.info(f"--> No {tel_name} events are found. Skipping...")
+                continue
+
+            logger.info(f"--> {n_events_magic} events are found.")
+
+            # Extract indexes of MAGIC large shower events
+            index_large_intensity_magic = np.argsort(
+                df_magic["intensity"][mask_lst_obs_window]
+            )[::-1][:n_pre_offset_search_events]
+
+            # If LST/MAGIC observations are not completely overlapped, only small
+            # numbers of MAGIC events are left for the pre offset search.
+            # To find large-intensity showers within the same time window,
+            # time cut around MAGIC observations is applied to the LST data set.
+            cond_lolim = (
+                timestamps_lst
+                >= timestamps_magic[mask_lst_obs_window][0] - window_half_width
+            )
+            cond_uplim = (
+                timestamps_lst
+                <= timestamps_magic[mask_lst_obs_window][-1] + window_half_width
+            )
+            mask_magic_obs_window = np.logical_and(cond_lolim, cond_uplim)
```

Comment:
just to make the code more "homogeneous", since you already reuse cond_lolim and cond_uplim, you may reassign above time_lolim and time_uplim to timestamps_magic[mask_lst_obs_window][0] - window_half_width and timestamps_magic[mask_lst_obs_window][-1] + window_half_width respectively